### PR TITLE
Adds notice support to the login command

### PR DESCRIPTION
**Pull request by arcanis**
**What's the problem this PR addresses?**

The npm registry is implementing some authentication features around tokens. For this to work they need us to display the `npm-notice` header.

Closes https://github.com/yarnpkg/berry/issues/5006

**How did you fix it?**

Prints the header before requesting the OTP token.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/plugin-npm/sources/npmHttpUtils.ts
Comment (merceyz) [inline packages/plugin-npm/sources/npmHttpUtils.ts:284]: Poor Windows users, we should just let https://yarnpkg.com/package/open handle this so we don't have to.

**Review by merceyz — CHANGES_REQUESTED** [review]

File: packages/plugin-npm/sources/npmHttpUtils.ts
Comment (paul-soporan) [reply packages/plugin-npm/sources/npmHttpUtils.ts:284]: Doesn't `open` spawn a bundled shell file, meaning that we can't bundle it?

Comment (merceyz) [reply packages/plugin-npm/sources/npmHttpUtils.ts:284]: It does but it will check if it can use it, if it can't it will use the system one.
https://github.com/sindresorhus/open/blob/05ba9e150cc1a2629e518a9cc19b586c6ca3f269/index.js#L173-L182

Comment (merceyz) [reply packages/plugin-npm/sources/npmHttpUtils.ts:284]: I think the better solution would be to not auto open the link, this seems like a CVE waiting to happen.

Comment (arcanis) [reply packages/plugin-npm/sources/npmHttpUtils.ts:284]: From a UX perspective I think the auto-open would still be very beneficial. What if we checked the hostname (or even the full url) to check it matches a known pattern? Various other tools have this kind of "open if possible" logic (AWS iirc?), so it should be possible to achieve it safely 🤔

Comment (arcanis) [reply packages/plugin-npm/sources/npmHttpUtils.ts]: I've added a prompt; pressing `<enter>` will try to open the url, but people can use `n` (or ctrl-c).

File: packages/yarnpkg-builder/sources/commands/build/bundle.ts
Comment (merceyz) [inline packages/yarnpkg-builder/sources/commands/build/bundle.ts:130]: Could you elaborate on this change / why are we including our test code in production now?

Comment (arcanis) [reply packages/yarnpkg-builder/sources/commands/build/bundle.ts:130]: Because tests use the bundle too; the login tests can't work at all currently. It's also a very small amount of code, with a reasonable use case for users (enable non-official programmatic usage), so I'm not sure requiring an extra `--no-tests` flag is really worth the very small size gain (110 bytes as of https://github.com/yarnpkg/berry/pull/4315).

Comment (merceyz) [reply packages/yarnpkg-builder/sources/commands/build/bundle.ts:130]: The tests use the un-minified build so that shouldn't affect our tests.
It's not really about the size, it's about not breaking users (test) setup if they set `TEST_ENV` and run `yarn`.
If we want to include our test code we should at least use a more specific name than `TEST_ENV` so that users don't accidentally break their setup.

Comment (arcanis) [reply packages/yarnpkg-builder/sources/commands/build/bundle.ts:130]: Ah, I see - I was talking about the local tests, which I often run after a simple `yarn build:cli`. I'm fine with using a Yarn-tagged env name, will make the change 👍

File: packages/plugin-npm/sources/npmHttpUtils.ts
Comment (merceyz) [reply packages/plugin-npm/sources/npmHttpUtils.ts]: I still don't think we should if we can't do it consistently.

Comment (arcanis) [reply packages/plugin-npm/sources/npmHttpUtils.ts]: Sorry, I saw your comment after merging - by consistently, do you refer to the Windows support? I can disable this altogether for win32, if you think that's better (I'm probably a little biased, but I'd like to avoid adding dependencies from this maintainer if I can avoid it ...).

Comment (merceyz) [reply packages/plugin-npm/sources/npmHttpUtils.ts]: Yeah, referring to Windows support.

**Pull request merged**
